packages/core-data/src/entity-provider.js

```diff
@@ -20,48 +20,48 @@ const EMPTY_ARRAY = [];
 /**
  * Internal dependencies
  */
-import { defaultEntities, kinds } from './entities';
+import { rootEntitiesConfig, additionalEntityConfigLoaders } from './entities';
 
-const entities = {
-	...defaultEntities.reduce( ( acc, entity ) => {
-		if ( ! acc[ entity.kind ] ) {
-			acc[ entity.kind ] = {};
+const entityContexts = {
+	...rootEntitiesConfig.reduce( ( acc, loader ) => {
+		if ( ! acc[ loader.kind ] ) {
+			acc[ loader.kind ] = {};
 		}
-		acc[ entity.kind ][ entity.name ] = { context: createContext() };
+		acc[ loader.kind ][ loader.name ] = { context: createContext() };
 		return acc;
 	}, {} ),
-	...kinds.reduce( ( acc, kind ) => {
-		acc[ kind.name ] = {};
+	...additionalEntityConfigLoaders.reduce( ( acc, loader ) => {
+		acc[ loader.kind ] = {};
 		return acc;
 	}, {} ),
 };
-const getEntity = ( kind, type ) => {
-	if ( ! entities[ kind ] ) {
+const getEntityContext = ( kind, name ) => {
+	if ( ! entityContexts[ kind ] ) {
 		throw new Error( `Missing entity config for kind: ${ kind }.` );
 	}
 
-	if ( ! entities[ kind ][ type ] ) {
-		entities[ kind ][ type ] = { context: createContext() };
+	if ( ! entityContexts[ kind ][ name ] ) {
+		entityContexts[ kind ][ name ] = { context: createContext() };
 	}
 
-	return entities[ kind ][ type ];
+	return entityContexts[ kind ][ name ].context;
 };
 
 /**
  * Context provider component for providing
- * an entity for a specific entity type.
+ * an entity for a specific entity.
  *
  * @param {Object} props          The component's props.
  * @param {string} props.kind     The entity kind.
- * @param {string} props.type     The entity type.
+ * @param {string} props.type     The entity name.
  * @param {number} props.id       The entity ID.
  * @param {*}      props.children The children to wrap.
  *
  * @return {Object} The provided children, wrapped with
  *                   the entity's context provider.
  */
-export default function EntityProvider( { kind, type, id, children } ) {
-	const Provider = getEntity( kind, type ).context.Provider;
+export default function EntityProvider( { kind, type: name, id, children } ) {
```

Comment:
It would be nice to refactor it at one point, but that's a part of the public API and beyond the scope of this PR.